Mage.Server/src/main/java/mage/server/UserManager.java

```diff
@@ -231,8 +231,8 @@ private void checkExpired() {
             }
             logger.debug("Users to remove " + toRemove.size());
             final Lock w = lock.readLock();
-            w.lock();
             try {
+                w.lock();
```

Comment:
Lock must be outside from own try-finally

Reply:
This was marked as not ensuring the lock would be unlocked in all execution paths, could you explain why the lock must be outside from that try finally?

Reply:
It's a wrong mark. If you failed with item.lock() then no need to call item.unlock(). Moreover if you try to unlock not locked item then it can raise exception (for some bad code/objects). It's a standard block with try-finally, [look here in javadocs as example](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/Lock.html).